### PR TITLE
Part2

### PR DESCRIPTION
**Pull request by morecallan**
@gib-bowden - Opening this up because it looks like you are ready for review. Can you confirm?

### CODE REVIEW

File: db/contacts.json [hunk 1]
```diff
@@ -0,0 +1,143 @@
+{
+	"contacts": [{
```
Comment (morecallan) [inline db/contacts.json]: This should be an object of objects with keys for each contact.

File: db/contacts.json [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+			"id": 1,
+			"firstName": "Gib",
+			"lastName": "Bowden",
+			"phone": "615-944-4766",
+			"email": "gibbowden@gmail.com",
+			"address": {
+				"street 1": "800 Woodland Street",
```
Comment (morecallan) [inline db/contacts.json]: You don't want nested objects with Firebase. You can either add them as single lines or create a new collection with of addresses with foreign keys on them. (I'd recommend going with the first option for this project).

File: javascripts/controllers/LoginCtrl.js [hunk 3]
```diff
@@ -1,5 +1,15 @@
 "use strict"; 
 
-app.controller("LoginCtrl", function($scope){
-    $scope.controller = "LoginCtrl"; 
+app.controller("LoginCtrl", function($location, $rootScope, $scope, AuthService){
+    $scope.authenticate = () => {
+        AuthService.authenticateGoogle().then((result) => {
+            console.log(result);
```
Comment (morecallan) [inline javascripts/controllers/LoginCtrl.js]: Remove console log.

File: db/contacts.json [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+				"street 2": "Unit 105",
+				"city": "Nashville",
+				"state": "TN",
+				"zip": "37206"
+			}
```
Comment (morecallan) [inline db/contacts.json]: You'll want to add a favorite key in here too because it is in your form.

**Review by morecallan — COMMENTED** [review]
@gib-bowden - Very well done. Looking pretty good. Make the suggested changes and add uid to the contacts.json contacts and push up and we'll merge it.

**Comment by gib-bowden** [comment]
@morecallan just now fixed this. ok to merge?

**Comment by morecallan** [comment]
👍

**Pull request merged**